### PR TITLE
add some form of validation

### PR DESCRIPTION
**Pull request by moh-moola**

### CODE REVIEW

File: molo/servicedirectory/tests/test_views.py [hunk 1]
```diff
@@ -136,8 +136,27 @@ def test_location_search_view(self):
...
+        }
+        response = self.client.get(
+            reverse('molo.servicedirectory:location-results'),
+            data=data
+        )
+
+        self.assertIn(
+            reverse('molo.servicedirectory:location-search'),
+            response.url,
+        )
```
Comment (KaitCrawford) [inline molo/servicedirectory/tests/test_views.py:152]: please check that the error message is in the response as well.

File: molo/servicedirectory/tests/test_views.py [hunk 2]
```diff
@@ -162,6 +181,15 @@ def test_location_results_view(self):
...
             response, 'servicedirectory/location_results.html')
 
+    def test_organisation_results_view_no_search_params(self):
+        response = self.client.get(
+            reverse('molo.servicedirectory:organisation-results')
+        )
+        self.assertIn(
+            reverse('molo.servicedirectory:home'),
+            response.url
+        )
```
Comment (KaitCrawford) [inline molo/servicedirectory/tests/test_views.py:195]: please check that the error message is in the response as well.

**Review by KaitCrawford — COMMENTED** [review]
Two requests. I'm also a little bit confused that you added 3 validation sections but only 2 tests. is the 3rd one tested somewhere else?

File: molo/servicedirectory/tests/test_views.py (hunk 2 above)
Comment (moh-moola) [reply molo/servicedirectory/tests/test_views.py:195]: :+1:

**Comment by KaitCrawford** [comment]
@moh-moola please wait for a proper approval before merging your PRs. I never approved or gave this PR a :+1: 
You haven't addressed my comment about ensuring you've tested all the functionality you added:
"I'm also a little bit confused that you added 3 validation sections but only 2 tests. is the 3rd one tested somewhere else?"

**Comment by moh-moola** [comment]
@KaitCrawford the issue here is that the code needs a refactor, one can bypass the steps, and hit the result pages, I need to change the views from a simple template view form views with step validation, this is just to ensure the minimum required parameters are always met. The third validation step just ensures that you have the values set at step1 as the values are only processed in the final step and this would be a an issue redirecting a user to the 1st step from the 3rd... I get your point I shall review a better way of handling this

**Pull request merged**